charts/external-dns/templates/deployment.yaml

```diff
@@ -126,7 +130,49 @@ spec:
           resources:
             {{- toYaml . | nindent 12 }}
           {{- end }}
-      {{- if or .Values.secretConfiguration.enabled .Values.extraVolumes }}
+        {{- if .Values.webhook.image }}
+        - name: webhook
+          {{- with .Values.securityContext }}
+          securityContext:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          image: {{ .Values.webhook.image . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- with .Values.webhook.env }}
+          env:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          args:
+          {{- range .Values.webhook.args }}
+            - {{ tpl . $ }}
+          {{- end }}
+          ports:
+            - name: webhook-metrics
+              protocol: TCP
+              containerPort: 8080
+          livenessProbe:
+            {{- toYaml .Values.webhook.livenessProbe | nindent 12 }}
+          readinessProbe:
+            {{- toYaml .Values.webhook.readinessProbe | nindent 12 }}
+          {{- if or .Values.webhook.secretConfiguration.enabled .Values.webhook.extraVolumeMounts }}
+          volumeMounts:
+            {{- if .Values.webhook.secretConfiguration.mountPath }}
+            - name: secrets
+              mountPath: {{ tpl .Values.webhook.secretConfiguration.mountPath $ }}
+            {{- with .Values.webhook.secretConfiguration.subPath }}
+              subPath: {{ tpl . $ }}
+            {{- end }}
+            {{- end }}
+            {{- with .Values.webhook.extraVolumeMounts }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
+          {{- end }}
+          {{- with .Values.webhook.resources }}
+          resources:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+        {{- end }}
+      {{- if or .Values.secretConfiguration.enabled .Values.webhook.secretConfiguration.enabled .Values.extraVolumes }}
       volumes:
         {{- if .Values.secretConfiguration.enabled }}
```

Comment:
```suggestion
        {{- if or .Values.secretConfiguration.enabled .Values.webhook.secretConfiguration.enabled }}
```

Reply:
I had decided against having a `.Values.webhook.secretConfiguration.enabled`. The `.Values.secretConfiguration.enabled` setting defines the creation of the Secret, whereas each of the containers has its own `mountPath` setting. It is most likely that any secret will only be needed by the webhook (to authenticate to the provider's API).